### ISSUE TITLE
Error in deployment: 'The github-pages gem can't satisfy your Gemfile's dependencies.'

### ISSUE THREAD

**Issue by cloybar** [alshedivat/al-folio#2542]
### Have you checked that your issue isn't already filed?

- [X] I read through [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my issue.
- [X] Yes, I have checked that this issue isn't already filed.

### Bug description

Webpage does not deploy

### How to reproduce the bug

I created a new proyect by forking the main al-folio proyect. Then, following the README.md, I sync one the local repo with the one of github, getting a deploy error. As it suggested, I ran `gem install bundler:2.5.7' and `bundle install` to update the gems. Since then, I am getting the same error, even though the gems that they say are missing, can be found on the Gemfile.

### Error messages and logs

```
Error:  Logging at level: debug Configuration file: /github/workspace/./_config.yml GitHub Pages: github-pages v231 GitHub Pages: jekyll v3.9.5 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-feed Requiring: jekyll-sitemap Requiring: jemoji Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-gist Requiring: jekyll-github-metadata Requiring: jekyll-paginate Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... EntryFilter: excluded /CONTRIBUTING.md EntryFilter: excluded /docker-compose-slim.yml EntryFilter: excluded /Gemfile EntryFilter: excluded /README.md EntryFilter: excluded /LICENSE EntryFilter: excluded /FAQ.md EntryFilter: excluded /Gemfile.lock EntryFilter: excluded /INSTALL.md EntryFilter: excluded /package.json EntryFilter: excluded /Dockerfile EntryFilter: excluded /package-lock.json EntryFilter: excluded /purgecss.config.js EntryFilter: excluded /CUSTOMIZE.md EntryFilter: excluded /docker-compose.yml Reading: _posts/2024-04-29-typograms.md Reading: _posts/2023-05-12-custom-blockquotes.md Reading: _posts/2023-04-24-videos.md Reading: _posts/2023-03-21-tables.md Reading: _posts/2015-10-20-disqus-comments.md Reading: _posts/2018-12-22-distill.md Reading: _posts/2020-09-28-github-metadata.md Reading: _posts/2023-12-12-tikzjax.md Reading: _posts/2015-03-15-formatting-and-links.md Reading: _posts/2022-12-10-giscus-comments.md Reading: _posts/2023-07-12-post-bibliography.md Reading: _posts/2023-04-25-sidebar-table-of-contents.md Reading: _posts/2024-01-27-advanced-images.md Reading: _posts/2024-01-26-geojson-map.md Reading: _posts/2020-09-28-twitter.md Reading: _posts/2015-10-20-math.md Reading: _posts/2023-03-20-table-of-contents.md Reading: _posts/2023-04-25-audios.md Reading: _posts/2024-01-27-code-diff.md Reading: _posts/2022-02-01-redirect.md Reading: _posts/2021-07-04-diagrams.md Reading: _posts/2015-05-15-images.md Reading: _posts/2024-04-28-post-citation.md Reading: _posts/2023-07-04-jupyter-notebook.md Reading: _posts/2024-01-26-chartjs.md Reading: _posts/2024-01-26-echarts.md Reading: _posts/2024-01-27-vega-lite.md Reading: _posts/2015-07-15-code.md Reading: _posts/2024-04-15-pseudocode.md Reading: _posts/2024-05-01-tabs.md EntryFilter: excluded /readme_preview/al-folio-preview.png EntryFilter: excluded /readme_preview/publications.png EntryFilter: excluded /readme_preview/photos-screenshot.png EntryFilter: excluded /readme_preview/math.png EntryFilter: excluded /readme_preview/dark.png EntryFilter: excluded /readme_preview/jupyter.png EntryFilter: excluded /readme_preview/people.png EntryFilter: excluded /readme_preview/code.png EntryFilter: excluded /readme_preview/blog.png EntryFilter: excluded /readme_preview/distill.png EntryFilter: excluded /readme_preview/cv.png EntryFilter: excluded /readme_preview/repositories.png EntryFilter: excluded /readme_preview/light.png EntryFilter: excluded /readme_preview/projects.png EntryFilter: excluded /lighthouse_results/mobile EntryFilter: excluded /lighthouse_results/desktop EntryFilter: excluded /_pages/about_einstein.md EntryFilter: excluded /bin/deploy EntryFilter: excluded /bin/entry_point.sh EntryFilter: excluded /bin/cibuild Reading: _news/announcement_2.md Reading: _news/announcement_3.md Reading: _news/announcement_1.md Reading: _projects/9_project.md Reading: _projects/5_project.md Reading: _projects/1_project.md Reading: _projects/7_project.md Reading: _projects/2_project.md Reading: _projects/3_project.md Reading: _projects/4_project.md Reading: _projects/6_project.md Reading: _projects/8_project.md Generating: JekyllOptionalFrontMatter::Generator finished in 6.3968e-05 seconds. Generating: JekyllReadmeIndex::Generator finis
```

```
build 
The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-a
```

### What operating system are you using?

Windows

EDIT: I have the same error in Github

### Where are you seeing the problem on?

Deployed site

### More info

I already had a deployed site following a video tutorial ([https://www.youtube.com/watch?v=g6AJ9qPPoyc](url)). So, I already had some of the stuff installed needed.

EDIT: I started from scratch and followed the recommended approach, but I get the exact same error for the deployment straight away after creating the repo and updating the file _config.yml

**Comment by george-gca** [comment]
You should [follow the recommended approach](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#recommended-approach) to the line. If you follow the recommended steps, you'll only do 2 changes in `_config.yml` (among with other configuration steps on GitHub) and the site should be live and working. Then, and only then, you try to do stuff on your machine.

**Comment by SwayStar123** [comment]
> You should [follow the recommended approach](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#recommended-approach) to the line. If you follow the recommended steps, you'll only do 2 changes in `_config.yml` (among with other configuration steps on GitHub) and the site should be live and working. Then, and only then, you try to do stuff on your machine and further customizations.

I have made no changes to the config other than changing the url to my username, and making baseurl blank, also followed all the settings instructions, but i still get that same error. I havent changed any other file

**Comment by george-gca** [comment]
Can you paste your error log again, but this time between 3 ticks? This way it is is unreadable, as it is a one liner.

**Comment by Siyuan-Shen** [comment]
Same error here.

**Comment by george-gca** [comment]
Install strictly following the [recommended approach](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#recommended-approach). Only AFTER the site is up and running you should clone and start modifying it.

**Comment by cloybar** [comment]
> You should [follow the recommended approach](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#recommended-approach) to the line. If you follow the recommended steps, you'll only do 2 changes in `_config.yml` (among with other configuration steps on GitHub) and the site should be live and working. Then, and only then, you try to do stuff on your machine and further customizations.

Thanks, I started from scratch. I installed WSL, however, I am getting an error deployment right after modifying the file _config.yml at a github level. I guess I should open another issue with this problem...

**Comment by cloybar** [comment]
> Can you paste your error log again, but this time between 3 ticks? This way it is is unreadable, as it is a one liner.
> 
> Edit: taking a quick look at it, it is stating about dependencies that we do not have. Are you sure you are using our template?

I am doing copy-paste of the error in between the 3 ticks, but it is shown like that (I tried to change it but there is no better display...).

And yes, now I am following [the recommended approach](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#recommended-approach) to the line, creating the new repo by clicking the link given there, and I still get the same error. I have no idea what the problem is.

**Comment by hansfarrell** [comment]
I believe I am encountering a similar issue as this, when I initially created the repository and the initial commit fails, particularly on the pages build and deployment workflow in the Actions tab. The workflow fails because 
``` Warning: The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/ ```
![Screenshot 2024-08-08 224725](https://github.com/user-attachments/assets/9ea82097-a22d-4a87-bde7-c94dc6ec0b59)

**Comment by george-gca** [comment]
This is a warning, so the build doesn't fail because of that. Check your full stacktrace for an `error` message.

**Comment by dylanmikesell** [comment]
I also had this issue. I think that in the **recommended approach** section you might want to be more explicit by what you mean by leave baseurl blank. I think many of us are doing the following.

`baseurl: / `

which builds, but it does not look right. The first time you build, you really need 
`baseurl: `

Then you if you change it to 
`baseurl: /`
or
`baseurl: `
everything is fine and the site has the correct theme. 

Thus, I am not sure if something extra happens when it is `baseurl: ` the first time. Maybe someone else discovered this too.

**Comment by george-gca** [comment]
We accept suggestion on how to better write this part. There are already a lot of issues and questions about the problem that is not setting correctly the `baseurl` and mostly of them already answered, but this seems to keep coming back.

**Comment by hansfarrell** [comment]
The problem is, after I first created a new repository based on this template and named it <my-username>.github.io, the initial commit of the repository fails, particularly in the Actions > pages build and deployment. Then, I changed the workflow permission based on the instruction to Read and Write permissions, but nothing changes. (The picture I attached is the Actions tab before I changed the workflow permission from the settings, but it doesn't change afterwards too)
![image](https://github.com/user-attachments/assets/a805ec52-a371-46d4-96ee-3cf68dc1fc00)

**Comment by KalebCole** [comment]
I am also getting this issue. I have followed the video tutorial outlined in INSTALL.md, where I make updates to config.yml through the GitHub browser. I made changes locally, but have since reset to the last commit from the forked repo.

I have tried both
base_url: /
base_url: 

Neither seems to work because I keep failing the GitHub actions build job in the workflow.

I get a dependency error in the logs

![image](https://github.com/user-attachments/assets/92f867b8-e8d9-409e-a8ff-091e894a4ee4)

**Comment by george-gca** [comment]
Your error is related to not setting the deployment branch correctly, as specified in the [FAQ](https://github.com/alshedivat/al-folio/blob/main/FAQ.md#my-webpage-works-locally-but-after-deploying-it-fails-to-build-and-throws-unknown-tag-toc-how-do-i-fix-that).

**Comment by KalebCole** [comment]
I agree that I would need the deployment branch to be gh-pages, but I'm confused on if I do this manually.

From my understanding of the [INSTALL.md,](https://github.com/alshedivat/al-folio/blob/main/INSTALL.md#installing-and-deploying), this job would be done for me in the GitHub Actions pipeline. But, I am failing the GitHub Actions pipeline that is supposed to do that for me (I think).

Do I need to manually create the branch and then set the deployment branch as gh-pages in GitHub?
